### PR TITLE
rpcdaemon: add ws compression

### PR DESCRIPTION
**Pull request by lupin012**

### CODE REVIEW

File: cmd/common/rpcdaemon_options.cpp [hunk 1]
```diff
@@ -111,6 +111,10 @@ void add_rpcdaemon_options(CLI::App& cli, silkworm::rpc::DaemonSettings& setting
     cli.add_flag("--websocket", settings.use_websocket)
         ->description("Enable WebSocket protocol for Execution Layer and Engine JSON RPC API, same port as HTTP(S)")
         ->capture_default_str();
+
+    cli.add_flag("--ws-compression", settings.ws_compression)
```
Comment (canepat) [inline cmd/common/rpcdaemon_options.cpp:115]: OK for this new flag, but then I suggest renaming `--websocket` flag as `--ws`

File: cmd/common/rpcdaemon_options.cpp [hunk 2]
```diff
@@ -111,6 +111,10 @@ void add_rpcdaemon_options(CLI::App& cli, silkworm::rpc::DaemonSettings& setting
...
+        ->description("Enable ws-compression protocol for Execution Layer and Engine JSON RPC API")
```
Comment (canepat) [inline cmd/common/rpcdaemon_options.cpp:116]: Please change description to "Enable compression on WebSocket protocol for Execution Layer and Engine JSON RPC API"

File: silkworm/rpc/ws/connection.cpp [hunk 3]
```diff
@@ -43,6 +45,12 @@ Connection::~Connection() {
 Task<void> Connection::accept(const boost::beast::http::request<boost::beast::http::string_body>& req) {
     // Set suggested timeout settings for the websocket
     ws_.set_option(boost::beast::websocket::stream_base::timeout::suggested(boost::beast::role_type::server));
+    if (ws_compression_) {
+        boost::beast::websocket::permessage_deflate opt;
+        opt.client_enable = true;
+        opt.server_enable = true;
```
Comment (canepat) [inline silkworm/rpc/ws/connection.cpp:51]: Better use designated initialisers for this aggregate:
```
boost::beast::websocket::permessage_deflate opt{
    .client_enable = true,
    .server_enable = true,
};
```

File: silkworm/rpc/ws/connection.hpp [hunk 4]
```diff
@@ -44,17 +44,18 @@ class Connection : public Channel {
     //! Construct a connection running within the given execution context.
     Connection(boost::beast::websocket::stream<boost::beast::tcp_stream>&& stream,
                commands::RpcApi& api,
-               const commands::RpcApiTable& handler_table);
+               const commands::RpcApiTable& handler_table,
+               bool ws_compression = false);
```
Comment (canepat) [inline silkworm/rpc/ws/connection.hpp:48]: `ws_compression` -> `compression`

File: silkworm/rpc/ws/connection.hpp [hunk 5]
```diff
@@ -64,10 +65,14 @@ class Connection : public Channel {
...
     Task<std::size_t> do_write(const std::string& content);
 
+    // websocket stream
     boost::beast::websocket::stream<boost::beast::tcp_stream> ws_;
 
     //! The handler used to process the incoming request.
     http::RequestHandler request_handler_;
+
+    //! enable compress flag
+    bool ws_compression_{false};
```
Comment (canepat) [inline silkworm/rpc/ws/connection.hpp:75]: `ws_compression_` -> `compression_`

**Review by canepat — CHANGES_REQUESTED** [review]

File: cmd/common/rpcdaemon_options.cpp (hunk 1 above)
Comment (lupin012) [reply cmd/common/rpcdaemon_options.cpp:115]: ok

File: cmd/common/rpcdaemon_options.cpp (hunk 2 above)
Comment (lupin012) [reply cmd/common/rpcdaemon_options.cpp:116]: ok

File: silkworm/rpc/ws/connection.cpp (hunk 3 above)
Comment (lupin012) [reply silkworm/rpc/ws/connection.cpp:51]: ok

File: silkworm/rpc/ws/connection.hpp (hunk 4 above)
Comment (lupin012) [reply silkworm/rpc/ws/connection.hpp:48]: ok

File: silkworm/rpc/ws/connection.hpp (hunk 5 above)
Comment (lupin012) [reply silkworm/rpc/ws/connection.hpp:75]: ok

**Pull request merged**